recipes/mosquitto/2.x/conandata.yml

```diff
@@ -1,4 +1,7 @@
 sources:
+  "2.0.7":
```

Comment:
Upstream added CMake option `WITH_CJSON` to guard a `find_package` call. It should be controlled in the recipe too.

Reply:
https://github.com/eclipse/mosquitto/commit/c16514ca23d02af4b82b85ce3dcfd3cf97d77db0#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20a

https://github.com/eclipse/mosquitto/commit/bbe0afbfbe7bb392361de41e275759ee4ef06b1c#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20a

Default is ON and I don't see how the consumer could access that 🤔 

If someone wants to disable it that is for a separate PR

Reply:
Please add it and backport this behavior in previous versions, we must have full control of dependencies used based on recipe option.

Reply:
Just add to config_options:

```
def config_options(self):
    if tools.Version(self.version) > "2.0.6":
        self.options["cjson"] = True
```

Config options runs before of evaluating command line options. Thus, you just change the default options

Reply:
Hmm. Its already an option, @uilianries comment makes the most sense.

I'm off to bed but I'll fix it tomorrow

Reply:
Here is the modification to avoid unpredictable binary:

```python
        if tools.Version(self.version) < "2.0.6":
            self._cmake.definitions["CMAKE_DISABLE_FIND_PACKAGE_cJSON"] = not self.options.get_safe("cjson")
        else:
            self._cmake.definitions["WITH_CJSON"] = self.options.get_safe("cjson")
```

For default value, I would just turn it to True unconditionally.

Reply:
@qchateau Can you please add me as a contributor to you CCI fork  so I can address the comments above

Reply:
Sent you an invite